tiledb/sm/array_schema/dimension.cc

```diff
@@ -399,40 +417,50 @@ Status Dimension::check_tile_extent() const {
 
   auto tile_extent = static_cast<T*>(tile_extent_);
```

Comment:
Why do we cast to `T` here? Shouldn't the tile extent always be > 0? In which case maybe uint64 is a better cast.

Reply:
Yes it is always `>0` (I added a check for that recently), but this prevents some warning I am getting with the computations. Currently the API assumes that `tile_extent` (as set by the user when creating a dimension) is always of the same type as the domain. If we make it `uint64_t`, we may have to change the API as well, and add some more checks.